api-docs/client-files.md

```diff
@@ -0,0 +1,262 @@
+# Client files
+
+This api is provided to upload and keep track of client files.
+
+## Get a single client file
+
+```http
+GET /api/clients/:file/:id/files
+```
+
+### Response
+
+#### Success
+
+```json
+{
+  "createdBy": {
+    "firstName": "Sean",
+    "lastName": "White",
+    "fullName": "Sean White",
+    "timestamp": "2019-11-06T06:00:00.000Z"
+  },
+  "lastUpdatedBy": {
+    "firstName": "Sean",
+    "lastName": "White",
+    "fullName": "Sean White",
+    "timestamp": "2019-11-06T06:00:00.000Z"
+  },
+  "fileName": "file-1",
+  "filePath": "file://host/path"
+}
+```
+
+#### Not Found
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+If there is no file with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find file with #123"]
+}
+```
+
+&nbsp;
+
+---
+
+&nbsp;
+
+## Get all client files
+
+```http
+GET /api/clients/:id/files
+```
+
+### Response
+
+#### Success
+
+The response for all files will look identical to get a single file but will return a array of files
+
+```json
+{
+  "files": [
+    {
+      "createdBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "lastUpdatedBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "fileName": "file-1",
+      "filePath": "file://host/path"
+    },
+    "..."
+  ]
+}
+```
+
+#### Not Found
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+&nbsp;
+
+---
+
+&nbsp;
+
+## Create a single client file
+
+```http
+POST /api/client/file
+```
+
+### Request
```

Comment:
Uploading files via a browser is only possible by using [multipart form data](https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects). This means that the request headers for uploading a file must include `Content-Type: multipart/form-data`, and the request body should be the multipart form data format (which is not JSON). We should clarify all of this in the API docs, and also remove the JSON you have here in the request body.

Unless, are you expecting the file upload to go to S3 and this POST request just to be the way the UI tells the web server that they completed the uploading of a file?